src/main/java/com/oneops/api/resource/Assembly.java

```diff
@@ -312,4 +285,136 @@ public CiResource deleteAssembly(String assemblyName) throws OneOpsClientAPIExce
 		String msg = String.format("Failed to delete assembly with name %s due to null response", assemblyName);
 		throw new OneOpsClientAPIException(msg);
 	}
+
+
+	/**
+	 * AddTeams for the given assembly
+	 *
+	 * @param orgName      {mandatory}
+	 * @param assemblyName {mandatory}
+	 * @param teams        {mandatory}
+	 * @return teamList
+	 * @throws OneOpsClientAPIException
+	 */
+	public List<CiResource> addTeamsByAssembly(String orgName, String assemblyName, List<String> teams) throws OneOpsClientAPIException {
+		List<CiResource> ciResourceList = null;
+		if (null!= teams && !teams.isEmpty()) {
+			List<Team> orgTeamList = listOrganizationTeams(orgName);
+			List<String> inputTeamList = new ArrayList<>();
+			Map<String, Object> orgTeamMap = new HashMap<>();
+			if (null != orgTeamList && !orgTeamList.isEmpty()) {
+				for (Team team : orgTeamList) {
+					orgTeamMap.put(team.getName(), team.getId());
+				}
+			}
+			for (String teamName : orgTeamMap.keySet()) {
+				if (teams.contains(teamName)) {
+					inputTeamList.add(String.valueOf(orgTeamMap.get(teamName)));
+				}
+			}
+			if (!inputTeamList.isEmpty()) {
+				ciResourceList = addingTeamNameByAssembly(orgName, assemblyName, inputTeamList);
+			}
+		}
+		return ciResourceList;
+	}
+
+	/**
+	 * List of teams for the given organization
+	 *
+	 * @param orgName {mandatory}
+	 * @return
+	 * @throws OneOpsClientAPIException
+	 */
+	public List<Team> listOrganizationTeams(String orgName) throws OneOpsClientAPIException {
+		if(orgName == null || orgName.length() == 0) {
+			String msg = "Missing organization name to fetch details";
+			throw new OneOpsClientAPIException(msg);
+		}
+		RequestSpecification request = createRequest();
+		Response response = request.get(IConstants.ORGANIZATION_URI + "teams");
+		if(response != null) {
+			if(response.getStatusCode() == 200) {
+				return JsonUtil.toObject(response.getBody().asString(), new TypeReference<List<Team>>(){});
+			} else {
+				String msg = String.format("Failed to get team list for organization with name %s due to %s", orgName, response.getStatusLine());
+				throw new OneOpsClientAPIException(msg);
+			}
+		}
+		String msg = String.format("Failed to get team list for organization with name %s due to null response", orgName);
+		throw new OneOpsClientAPIException(msg);
+	}
+	/**
+	 * Adding Team Name By Assembly
+	 *
+	 * @param orgName {mandatory}
+	 * @param assemblyName {mandatory}
+	 * @param inputTeamList {mandatory}
+	 * @return
+	 * @throws OneOpsClientAPIException
+	 */
+	public List<CiResource> addingTeamNameByAssembly(String orgName, String assemblyName, List<String> inputTeamList) throws OneOpsClientAPIException {
+		if (assemblyName == null || assemblyName.length() == 0) {
+			String msg = "Missing assembly name to create teams for assembly";
```

Comment:
Please modify as `Assembly name is mandatory to add teams to assembly`, because we are not creating teams here but just adding existing teams to assembly.